utils/optimize.py

```diff
@@ -631,51 +668,37 @@ def optimize(filename):
 		# Iterate over the lines
 		i = 0
 		while i < n:
-			text = lines[i]
-			# Remove comments
-			parts = text.split(';', 1)
-			code = parts[0].rstrip()
-			comment = parts[1].strip() if len(parts) > 1 else ''
-			# Skip blank lines:
-			if not code:
-				i += 1
-				continue
+			cur_line = lines[i]
 			# Save the most recent label for context
-			if (code[0].isalpha() or code[0] == '_') and ':' in code:
-				cur_label = Line(i+1, code, comment, text, code)
-			# Remove indentation from code, if any
-			code = code.lstrip()
-			# Normalize whitespace
-			code = re.sub(r'\s+', ' ', code)
-			# Record the line's properties
-			context = cur_label.code if cur_label else ''
-			cur_line = Line(i+1, code, comment, text, context)
+			if (':' in cur_line.code and (cur_line.code[0].isalpha() or cur_line.code[0] == '_')
+					and not cur_line.text[0].isspace()):
+				cur_label = cur_line
 			# Check the condition for the current state
 			condition = conditions[state]
-			allow_rewind = type(condition) == tuple
+			allow_rewind = isinstance(condition, tuple)
 			if allow_rewind:
 				rewind_count, condition = condition
-			skip = comment.lower().startswith(SUPPRESS + ' ' + pattern_name.lower())
-			if condition(cur_line, prev_lines) and not skip:
+			skip = cur_line.comment_lower.startswith(SUPPRESS + pattern_name.lower())
+			if not skip and condition(cur_line, prev_lines):
 				# The condition was met; advance to the next state
 				prev_lines.append(cur_line)
 				state += 1
 				if state == len(conditions):
 					# All the conditions were met; print the result and reset the state
 					count += 1
 					if not printed_this and len(patterns) > 1:
-						print('###', pattern_name, '###')
+						print(f'### {pattern_name} ###')
+						printed_this = True
 					if cur_label:
-						prev_lines.insert(0, cur_label)
+						print(f'{filename}:{cur_label.num}:{cur_label.text}')
 					for line in prev_lines:
-						print(filename, line.num, line.text, sep=':')
+						print(f'{filename}:{line.num}:{line.text}')
 					printed = True
-					printed_this = True
 					prev_lines = []
 					state = 0
 			elif allow_rewind and not skip:
 				# The condition was not met; go back to a previous condition
-				i -= 1
+				i -= rewind_count
```

Comment:
I think this is supposed to be 1, but can't find a codebase that would trigger it right now.